pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -129,47 +124,39 @@ type NodeInfo struct {
 }
 
 // New creates an instance of a mon cluster
-func New(
-	clusterInfo *cephconfig.ClusterInfo,
-	context *clusterd.Context,
-	namespace, dataDirHostPath, rookVersion string,
-	cephVersion cephv1.CephVersionSpec,
-	mon cephv1.MonSpec,
-	placement rookalpha.Placement,
-	hostNetwork bool,
-	resources v1.ResourceRequirements,
-	ownerRef metav1.OwnerReference,
-) *Cluster {
+func New(context *clusterd.Context, namespace, dataDirHostPath string, hostNetwork bool, ownerRef metav1.OwnerReference) *Cluster {
 	return &Cluster{
-		clusterInfo:          clusterInfo,
-		context:              context,
-		placement:            placement,
-		dataDirHostPath:      dataDirHostPath,
-		Namespace:            namespace,
-		rookVersion:          rookVersion,
-		cephVersion:          cephVersion,
-		Count:                mon.Count,
-		AllowMultiplePerNode: mon.AllowMultiplePerNode,
-		maxMonID:             -1,
-		waitForStart:         true,
-		monPodRetryInterval:  6 * time.Second,
-		monPodTimeout:        5 * time.Minute,
-		monTimeoutList:       map[string]time.Time{},
-		HostNetwork:          hostNetwork,
+		context:             context,
+		dataDirHostPath:     dataDirHostPath,
+		Namespace:           namespace,
+		maxMonID:            -1,
+		waitForStart:        true,
+		monPodRetryInterval: 6 * time.Second,
+		monPodTimeout:       5 * time.Minute,
+		monTimeoutList:      map[string]time.Time{},
+		HostNetwork:         hostNetwork,
 		mapping: &Mapping{
 			Node: map[string]*NodeInfo{},
 			Port: map[string]int32{},
 		},
-		resources: resources,
```

Comment:
Why did you remove that? It's needed, I'm going to include it again in: https://github.com/rook/rook/pull/2764

Reply:
Instead of a `resources` var on the `Cluster` spec, now we can just access it with the `spec` directly from the `ClusterSpec` type. See the usage [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/mon/spec.go#L153).